src/app/settings/components/project-role-edit-panel/project-role-jwt-tokens.tsx

```diff
@@ -0,0 +1,143 @@
+
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import { FormApi, Text } from 'react-form';
+
+import { FormField } from '../../../shared/components';
+import { AppContext } from '../../../shared/context';
+import { JwtToken } from '../../../shared/models';
+import { CreateJWTTokenParams, DeleteJWTTokenParams, services } from '../../../shared/services';
+
+interface ProjectRoleJWTTokensProps {
+    tokens: JwtToken[];
+    api: FormApi;
+}
+
+export class ProjectRoleJWTTokens extends React.Component<ProjectRoleJWTTokensProps, any> {
+    public static contextTypes = {
```

Comment:
React is going to deprecate context api. Please use context `Consumer` instead. Here is example:

https://github.com/argoproj/argo-cd-ui/blob/40fdda3f5a6b71fdf22c295c1bc0205b95f51307/src/app/shared/components/query.tsx#L5-L7

Reply:
fixed